aiohttp/web.py

```diff
@@ -289,16 +290,30 @@ def run_app(app, *, host='0.0.0.0', port=None,
                                **make_handler_kwargs)
 
     loop.run_until_complete(app.startup())
-    srv = loop.run_until_complete(loop.create_server(handler, host,
-                                                     port, ssl=ssl_context,
-                                                     backlog=backlog))
 
     scheme = 'https' if ssl_context else 'http'
-    base_url = URL('{}://localhost'.format(scheme)).with_port(port)
-    if isinstance(host, str):
-        urls = [base_url.with_host(host)]
+
+    if path is None:
+        # Assume IP networking
+        srv_creation = loop.create_server(
+            handler, host, port, ssl=ssl_context, backlog=backlog
+        )
+        base_url = URL('{}://localhost'.format(scheme)).with_port(port)
+        if isinstance(host, str):
+            urls = (base_url.with_host(host),)
+        else:
+            urls = [base_url.with_host(h) for h in host]
     else:
-        urls = [base_url.with_host(h) for h in host]
+        srv_creation = loop.create_unix_server(
+            handler, path, ssl=ssl_context, backlog=backlog
+        )
+        # nginx-style URLs for Unix socket paths
+        if isinstance(path, str):
+            urls = ('{}://unix:{}:'.format(scheme, path),)
+        else:
+            urls = ['{}://unix:{}:'.format(scheme, p) for p in path]
+
+    srv = loop.run_until_complete(srv_creation)
```

Comment:
Doesn't sound bad. I will have to consider the impact on the parameters (for example, when assuming a host of '0.0.0.0')